src/coreclr/jit/codegenloongarch64.cpp

```diff
@@ -7068,7 +7117,18 @@ void CodeGen::genCall(GenTreeCall* call)
         }
         else if (abiInfo.IsSplit())
         {
-            NYI("unimplemented on LOONGARCH64 yet");
+            assert(compFeatureArgSplit());
+            genConsumeArgSplitStruct(argNode->AsPutArgSplit());
+
+            regNumber argReg   = (regNumber)((unsigned)abiInfo.GetRegNum() + idx);
+            regNumber allocReg = argNode->AsPutArgSplit()->GetRegNumByIdx(idx);
+
+            // For LA64's ABI, the split is only using the A7 and stack for passing arg.
+            assert(emitter::isGeneralRegister(argReg));
+            assert(emitter::isGeneralRegister(allocReg));
+            assert(abiInfo.NumRegs == 1);
+
+            inst_Mov(TYP_I_IMPL, argReg, allocReg, /* canSkip */ true);
```

Comment:
It does not look right to use `TYP_I_IMPL` here for `inst_Mov`. What if the argument is of a GC type (e. g. `struct { object, long }`?

There is a similar problem below in the single-reg argument case.

Reply:
Thanks very much.